lib/Models/WebMapTileServiceCatalogItem.ts

```diff
@@ -0,0 +1,657 @@
+    const resourceUrl: ResourceUrl | ResourceUrl[] | undefined =
+      layer.ResourceURL;
+    let baseUrl: string = new URI(this.url).search("").toString();
+    if (resourceUrl) {
+      if (Array.isArray(resourceUrl)) {
+        for (let i = 0; i < resourceUrl.length; i++) {
+          const url: ResourceUrl = resourceUrl[i];
+          if (
+            url.format.indexOf(format) !== -1 ||
+            url.format.indexOf("png") !== -1
+          ) {
+            baseUrl = url.template;
+          }
+        }
+      } else {
+        if (
+          format === resourceUrl.format ||
+          resourceUrl.format.indexOf("png") !== -1
+        ) {
+          baseUrl = resourceUrl.template;
+        }
+      }
+    }
+
+    const tileMatrixSet = this.tileMatrixSet;
+    if (!isDefined(tileMatrixSet)) {
+      return;
+    }
+
+    let rectangle;
+
+    if (
+      this.rectangle !== undefined &&
+      this.rectangle.east !== undefined &&
+      this.rectangle.west !== undefined &&
+      this.rectangle.north !== undefined &&
+      this.rectangle.south !== undefined
+    ) {
+      rectangle = Rectangle.fromDegrees(
+        this.rectangle.west,
+        this.rectangle.south,
+        this.rectangle.east,
+        this.rectangle.north
+      );
+    } else {
+      rectangle = undefined;
+    }
+
+    const imageryProvider = new WebMapTileServiceImageryProvider({
+      url: proxyCatalogItemUrl(this, baseUrl),
+      layer: layerIdentifier,
+      style: this.style,
+      tileMatrixSetID: tileMatrixSet.id,
+      tileMatrixLabels: tileMatrixSet.labels,
+      minimumLevel: tileMatrixSet.minLevel,
+      maximumLevel: tileMatrixSet.maxLevel,
+      tileWidth: tileMatrixSet.tileWidth,
+      tileHeight: tileMatrixSet.tileHeight,
+      tilingScheme: new WebMercatorTilingScheme(),
+      format,
+      rectangle
+    });
+    return imageryProvider;
+  }
+
+  @computed
+  get tileMatrixSet():
+    | {
+        id: string;
+        labels: string[];
+        maxLevel: number;
+        minLevel: number;
+        tileWidth: number;
+        tileHeight: number;
+      }
+    | undefined {
+    const stratum = <WmtsCapabilitiesStratum>(
+      this.strata.get(GetCapabilitiesMixin.getCapabilitiesStratumName)
+    );
+    if (!this.layer) {
+      return;
+    }
+    const layer = stratum.capabilitiesLayer;
+    if (!layer) {
+      return;
+    }
+
+    const usableTileMatrixSets = stratum.usableTileMatrixSets;
+
+    let tileMatrixSetLinks: TileMatrixSetLink[] = [];
+    if (layer?.TileMatrixSetLink) {
+      if (Array.isArray(layer?.TileMatrixSetLink)) {
+        tileMatrixSetLinks = [...layer?.TileMatrixSetLink];
+      } else {
+        tileMatrixSetLinks = [layer.TileMatrixSetLink];
+      }
+    }
+
+    let tileMatrixSetId: string =
+      "urn:ogc:def:wkss:OGC:1.0:GoogleMapsCompatible";
+    let maxLevel: number = 0;
+    let minLevel: number = 0;
+    let tileWidth: number = 256;
+    let tileHeight: number = 256;
```

Comment:
If a layer is added which doesn't support WebMercator, Terria will crash when it is added to the map. My guess is it has something to do with these default values.
Would it be worth throwing an error if no usable tile matrix can be found - or do you need these defaults for certain WMTS servers?

Reply:
it was done this way in master, I tested with few layers that support only GeographicTilingScheme and Terria didn't crash there was just no layer on the map (although there was no error that it failed to load). For me, it would be much better to remove those defaults and throw an error (just maybe drop a comment in code that there were defaults if we run into a server that by default works with the above matrix set and doesn't return it - never saw one indeed).